src/test_l3_tvSettings.c

```diff
@@ -0,0 +1,5224 @@
+    readInt(&durationValue);
+
+    // Validate the input values
+    if (fromValue < 0 || fromValue > 100)
+    {
+        UT_LOG_ERROR("Invalid 'from' backlight value entered. Exiting test.");
+        UT_LOG_INFO("Out %s", __FUNCTION__);
+        return;
+    }
+
+    if (toValue < 0 || toValue > 100)
+    {
+        UT_LOG_ERROR("Invalid 'to' backlight value entered. Exiting test.");
+        UT_LOG_INFO("Out %s", __FUNCTION__);
+        return;
+    }
+
+    if (durationValue < 0 || durationValue > 1000)
+    {
+        UT_LOG_ERROR("Invalid fade duration entered. Exiting test.");
+        UT_LOG_INFO("Out %s", __FUNCTION__);
+        return;
+    }
+
+    // Set the backlight fade settings
+    UT_LOG_INFO("Calling SetBacklightFade(IN:from:[%d], IN:to:[%d], IN:duration:[%d] ms)", fromValue, toValue, durationValue);
+    ret = SetBacklightFade(fromValue, toValue, durationValue);
+    UT_LOG_INFO("Result SetBacklightFade(IN:from:[%d], IN:to:[%d], IN:duration:[%d] ms, tvError_t:[%s])", fromValue, toValue, durationValue, ut_control_GetMapString(tvError_mapTable, ret));
+    ASSERT(ret == tvERROR_NONE);
+
+    // Get the backlight fade settings to confirm
+    UT_LOG_INFO("Calling GetCurrentBacklightFade(OUT:from:[], OUT:to:[], OUT:current:[])");
+    ret = GetCurrentBacklightFade(&currentFrom, &currentTo, &currentProgress);
+    UT_LOG_INFO("Result GetCurrentBacklightFade(OUT:from:[%d], OUT:to:[%d], OUT:current:[%d], tvError_t:[%s])", currentFrom, currentTo, currentProgress, ut_control_GetMapString(tvError_mapTable, ret));
+    ASSERT(ret == tvERROR_NONE);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test sets the current backlight mode.
+*
+* This test function retrieves the supported backlight modes, sets each supported mode,
+* and then verifies that the mode is correctly applied by getting the current backlight mode.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 09@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [tvSettings_L3_Low-Level_TestSpecification.md](../docs/pages/tvSettings_L3_Low-Level_TestSpecification.md)
+*/
+void test_l3_tvSettings_backlightMode(void)
+{
+    gTestID = 9;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    tvError_t ret = tvERROR_NONE;
+    int userChoice = 0;
+    int selectedMode = 0;
+    tvBacklightMode_t currentMode = tvBacklightMode_INVALID;
+    int menuIndex = 1;
+
+    // Display available modes to the user
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("\t\tSupported Backlight Modes");
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("\t#   %-30s", "Backlight Modes");
+    for (int i = tvBacklightMode_MANUAL; i < tvBacklightMode_MAX; i <<= 1)
+    {
+        UT_LOG_INFO("%d: [%s]", menuIndex++, ut_control_GetMapString(tvBacklightMode_mapTable, i));
```

Comment:
instead of menuIndex, directly use i (i.e enum value), check other places. Use the actual value of enum/item as menu index

Reply:
enum is powers of 2

typedef enum
{
    tvDataColor_NONE= 0,                        //!< No color
    tvDataColor_RED = 1,                        //!< Color is Red
    tvDataColor_GREEN = 2,                      //!< Color is Green
    tvDataColor_BLUE = 4,                       //!< Color is Blue
    tvDataColor_YELLOW = 8,                     //!< Color is Yellow
    tvDataColor_CYAN = 16,                      //!< Color is Cyan
    tvDataColor_MAGENTA = 32,                   //!< Color is Megenta
    tvDataColor_MAX = 64                        //!< End of enum
}tvDataComponentColor_t;